distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Ordering/Natural_Order.enso

```diff
@@ -0,0 +1,69 @@
+from Standard.Base import all
+
+import Standard.Base.Data.Text.Regex
+import Standard.Base.Data.Text.Regex.Mode
+import Standard.Base.Data.Ordering.Vector_Lexicographic_Order
+polyglot java import org.enso.base.Regex_Utils
+
+## Compares two text values according to the natural dictionary ordering.
+
+   > Example
+     Compare two texts according to the natural dictionary ordering.
+
+         Natural_Order.compare "a2" "a100" == Ordering.Less
+
+   > Example
+     Sort a vector of texts according to the natural dictionary ordering.
+
+         ["a2", "a1", "a100", "a001", "a0001"].sort by=Natural_Order.compare . should_equal ["a0001", "a001", "a1", "a2", "a100"]
+compare : Text -> Text -> Ordering
+compare text1 text2 =
+    nat1 = here.to_natural_key text1
+    nat2 = here.to_natural_key text2
+    ## TODO [RW] for some reason removing this `res` and returning the result
+       directly will make Vector.sort behave badly. This needs to be investigated
+       and filed as a bug, then this comment can be updated with a reference to a
+       relevant ticket.
+    res = Vector_Lexicographic_Order.compare nat1 nat2
+    res
+
+## PRIVATE
+to_natural_key : Text -> Vector
+to_natural_key text = if text.is_empty then [] else
+    splitter_regex = "([^0-9]+|[0-9]+)"
+    # TODO [RW] should change to the other implementation once Regex is fixed
```

Comment:
What does it mean "once Regex is fixed"? All comments should be clear to the reader. I do not understand what "fixed" means. Please explain the problem with Regexp better. Also, link an issue here.

Reply:
Got it fixed within this PR, so the comment is removed.